x-pack/plugins/endpoint/server/routes/endpoints.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema } from '@kbn/config-schema';
+import { EndpointAppContext, EndpointData } from '../types';
+import { AllEndpointsQueryBuilder } from '../services/endpoint/endpoint_query_builders';
+
+interface HitSource {
+  _source: EndpointData;
+}
+
+export interface EndpointResultList {
+  endpoints: EndpointData[];
+  total: number;
+  requestPageSize: number;
+  requestIndex: number;
+}
+
+export function registerEndpointRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  router.get(
+    {
+      path: '/api/endpoint/endpoints',
+      validate: {
+        query: schema.object({
+          pageSize: schema.number({ defaultValue: 10 }),
+          pageIndex: schema.number({ defaultValue: 0 }),
+        }),
+      },
+      options: { authRequired: true },
+    },
+    async (context, req, res) => {
```

Comment:
Question/DIscussion: What are the pros for using an anonymous function like this? I definitely prefer having functions named for readability purposes.

This isn't too bad because you're passing in `router` and only one route is being added at once, but when there are multiple routes being added it's quite hard to read IMHO.

Reply:
> This isn't too bad because you're passing in router and only one route is being added at once, but when there are multiple routes being added it's quite hard to read IMHO.

This handler function is only related to this route so there is the documentation. In this case I think the name may be redundant.

Reply:
FYI: personally, I also like the naming on the function. One drawback of anonymous functions is debugging where you just see "anonymous" in stack traces or when doing heap/cpu profiling (but perhaps we don't do much of that, so its ok)